src/array_api_extra/_funcs.py

```diff
@@ -548,3 +559,279 @@ def sinc(x: Array, /, *, xp: ModuleType | None = None) -> Array:
+        else:
+            msg = f"Invalid value for copy: {copy!r}"  # type: ignore[unreachable]  # pyright: ignore[reportUnreachable]
+            raise ValueError(msg)
+
+        if copy:
+            try:
+                at_ = x.at
+            except AttributeError:
+                # Emulate at[] behaviour for non-JAX arrays
+                # with a copy followed by an update
+                if xp is None:
+                    xp = array_namespace(x)
+                # Create writeable copy of read-only numpy array
+                x = xp.asarray(x, copy=True)
+                if writeable is False:
+                    # A copy of a read-only numpy array is writeable
+                    writeable = None
+            else:
+                # Use JAX's at[] or other library that with the same duck-type API
+                args = (y,) if y is not _undef else ()
+                return getattr(at_[self.idx], at_op)(*args, **kwargs), None
+
+        if _is_update:
+            if writeable is None:
+                writeable = is_writeable_array(x)
+            if not writeable:
+                # sparse crashes here
+                msg = f"Array {x} has no `at` method and is read-only"
+                raise ValueError(msg)
+
+        return None, x
+
+    def get(self, **kwargs: Untyped) -> Untyped:
+        """Return ``x[idx]``. In addition to plain ``__getitem__``, this allows ensuring
+        that the output is either a copy or a view; it also allows passing
+        keyword arguments to the backend.
+        """
+        if kwargs.get("copy") is False:
+            if is_array_api_obj(self.idx):
+                # Boolean index. Note that the array API spec
+                # https://data-apis.org/array-api/latest/API_specification/indexing.html
+                # does not allow for list, tuple, and tuples of slices plus one or more
+                # one-dimensional array indices, although many backends support them.
+                # So this check will encounter a lot of false negatives in real life,
+                # which can be caught by testing the user code vs. array-api-strict.
+                msg = "get() with an array index always returns a copy"
+                raise ValueError(msg)
+            if is_dask_array(self.x):
+                msg = "get() on Dask arrays always returns a copy"
+                raise ValueError(msg)
+
+        res, x = self._common("get", _is_update=False, **kwargs)
+        if res is not None:
+            return res
+        assert x is not None
+        return x[self.idx]
+
+    def set(self, y: Array, /, **kwargs: Untyped) -> Array:
+        """Apply ``x[idx] = y`` and return the update array"""
+        res, x = self._common("set", y, **kwargs)
+        if res is not None:
+            return res
+        assert x is not None
+        x[self.idx] = y
+        return x
+
+    def _iop(
+        self,
+        at_op: str,
+        elwise_op: Callable[[Array, Array], Array],
```

Comment:
since `operator.add` is passed here, maybe this shouldn't be `(Array, Array) -> Array`, but something like `(Scalar, Scalar) -> Scalar` 🤔

Reply:
x and y are always Array and this is a well-contained internal method; no need to make it as generic as possible

Reply:
I'm not talking about making it generic 🤔

Reply:
well, anything works with `operator.add` apparently 😅 

https://github.com/python/typeshed/blob/b0768596d2f7b2c12aa61d8a9714325284410b4b/stdlib/_operator.pyi#L55